docs/src/starknet/fees-and-versions.md

```diff
@@ -10,6 +10,11 @@ This is done using the `--max-gas` and `--max-gas-unit-price` flags.
 Cast allows you to specify either the version of the transaction you want to send or the fee token you want to pay the fees in. This is done using
 the `--version` and `--fee-token` flags.
 
+> ⚠️ **Warning**
+> `--fee-token` flag will be deprecated in the future due to [SNIP-16: Deprecation of Transaction Versions 0,1,2](https://community.starknet.io/t/snip-16-deprecation-of-transaction-versions-0-1-2/114443).
+> It is recommended to use `--version` flag instead or let the version be inferred automatically.
```

Comment:
Can we add this as a deprecation warning in the runtime as well? 
Show it when someone passes the `--fee-token` arg but have it display two variants:
* `--fee-token strk` just show warning that `--fee-token` will be removed
* `--fee-token eth` show warning that `--fee-token` will be removed & that eth transactions will not be supported in the future

Reply:
<img width="902" alt="image" src="https://github.com/user-attachments/assets/955642c3-5e8b-4abc-acf8-1a1f9b812293">
<img width="1065" alt="image" src="https://github.com/user-attachments/assets/177635b2-7713-4c62-88d7-82b357c97342">
without flag:
<img width="955" alt="image" src="https://github.com/user-attachments/assets/915d3ff5-808d-4084-81ec-0c1c14b1ff0c">